app/controllers/application_controller.rb

```diff
@@ -5,6 +5,138 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
+  get '/' do 
+    erb :index
+  end
+
+  get '/signup' do 
+    if session[:user_id]
+      redirect to 'tweets'
+    else
+      erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if params["username"].empty? || params["password"].empty? || params["email"].empty?
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.